c10/util/UniqueVoidPtr.h

```diff
@@ -67,6 +67,12 @@ class UniqueVoidPtr {
     return std::move(ctx_);
   }
 
+  C10_NODISCARD bool compare_exchange_deleter(DeleterFnPtr expected_deleter, DeleterFnPtr new_deleter) {
+    if (get_deleter() == expected_deleter) return false;
```

Comment:
I think this should be `!=`, not `==`.

Also, I'm wondering why this mistake didn't break any test cases, maybe improve test coverage?

Reply:
Answer is, we're still waiting on the CUDA tests to come back. I saw this fail very obviously on an internal flow I tested on.